api/ws/rpc/connect.js

```diff
@@ -159,13 +160,7 @@ const connectSteps = {
 	},
 
 	registerSocketListeners: (peer, logger, onDisconnectCb = () => {}) => {
-		// Unregister all the events just in case
 		const socket = peer.socket;
-		socket.off('connect');
```

Comment:
I believe we need to remove the event listeners if we delete this code we might get other errors, later on, so removing event listeners don't cause any side effects.

Reply:
We don't need to remove the event listeners anymore because now that `multiplex` is false, it will always create a new socket object so the socket will never have any event handlers attached at this point in the code so calling `off` here is wasteful and it opens up the possibility that an 'error' event on the socket could be uncaught and crash the worker process.  It's possible that the `socket.off('error')` call here is responsible for this issue: https://github.com/LiskHQ/lisk/issues/1875

All event handlers are attached to the SCSocket itself so they will be all be garbage collected at the same time as the SCSocket itself is garbage collected (when it.